tools/file-marker/marker.ts

```
@@ -0,0 +1,46 @@
+  readonly [FILE_MARKER]: true;
+  readonly generator: string;
+  readonly generatedAt: string;
+  readonly version: string;
+}
+
+/**
+ * Helper to create metadata for generated files
+ */
+export function createFileMetadata(
+  generator: string,
+  version: string = '1.0.0'
```

ESLint

[error] Type string trivially inferred from a string literal, remove type annotation.

(@typescript-eslint/no-inferrable-types)